src/Patient/PatientFunctions/PatientMakeAppointments/PatientMakeAppointments.js

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import CssBaseline from '@mui/material/CssBaseline';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { useState, useEffect } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { DateCalendar, PickersDay } from "@mui/x-date-pickers";
+import dayjs from 'dayjs';
+import './NurseMakeAppointments.css';
+
+const defaultTheme = createTheme();
+
+export default function PatientMakeAppointments() {
+    const { patient_id } = useParams();
+    const [selectedDate, setSelectedDate] = useState(null);
+    const [gender, setGender] = useState('');
+    const [appointments, setAppointments] = useState({});
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        // Fetch patient gender
+        const fetchPatientData = async () => {
+            try {
+                const response = await fetch(`http://127.0.0.1:3000/patients/find_by_patient_id?patient_id=${patient_id}`);
+                const data = await response.json();
+                setGender(data.gender);
+            } catch (error) {
+                console.error('Error fetching patient data:', error);
+            }
+        };
+
+        fetchPatientData();
+    }, [patient_id]);
+
+    const handleDateChange = async (date) => {
+        setSelectedDate(date);
+        if (gender) {
+            try {
+                const formattedDate = dayjs(date).format('YYYY-MM-DD');
+                const response = await fetch(`http://127.0.0.1:3000/appointments/find_by_date_and_patient_gender?date=${formattedDate}&gender=${gender}`);
```

Comment:
Lets try to take the host part of the url (including the protocol and port) to a common config. So that when you host the app you don't have to change the URL everywhere.

    http://127.0.0.1:3000